src/core/integrations/ApplePay.ts

```
@@ -338,7 +338,10 @@ class ApplePay {
     if (walletsession) {
       this.merchantSession = JSON.parse(walletsession);
       this.validateMerchantRequestData.walletmerchantid = this.merchantSession.merchantIdentifier;
-      this.session.completeMerchantValidation(this.merchantSession);
+      this.setNotification(NotificationType.Success, Language.translations.PAYMENT_SUCCESS);
+      if (this.session) {
+        this.session.completeMerchantValidation(this.merchantSession);
```

It looks like this would display a success message but then silently fail to continue if the session doesn't exist, does onValidateMerchantResponseFailure need to be called if the session didn't exist here?

---

this was dedicated for testing purposes (if we launch tests on some other browsers than Safari) but actually this test case doesn;t make sense.
I deleted this line.